### DISCUSSION TITLE
Is there any official email support?

### DISCUSSION THREAD

**Discussion by sudharsan-Trinity** [dashingsoft/pyarmor#1404] [General]
Hi Pyamor team. currently we are looking for tool to encrypt our source code. we shortlisted pyarmor because of the features it has will serve our purpose. im facing issue in encrypting using outer and i have few more question. i search the document to get the support email to connect with pyarmor team to clear our doubts before purchase. it will be really helpful if i got some poc to connect with to clear all our doubts

**Comment by sudharsan-Trinity** [comment]
it will be better if i got some official support mail id to connect with

**Comment by jondy** [comment]
Sorry, now email doesn't provide technical support.
Please first check 
http://pyarmor.readthedocs.io/en/stable/questions.html

And have a glance on documentation. 

If still have question, please ask here with clear title.

**Comment by sudharsan-Trinity** [comment]
We want to encrypt our API source code which is build on Django. is their any method or hook to show the script expired as a api response ? and is their any workaround we can do to make the license uploaded by the UI

**Comment by jondy** [comment]
> is their any method or hook to show the script expired as a api response ? 

No direct method. You need write runtime hook to check runtime key and show something by yourself, refer to
https://pyarmor.readthedocs.io/en/stable/reference/environments.html#hooks

And some other use cases
https://pyarmor.readthedocs.io/en/stable/tutorial/customization.html

>  and is their any workaround we can do to make the license uploaded by the UI

If the uploaded license is generated by yourself with Pyarmor, it's just a simple file, you can do anything as a normal file.
